tests/testArgumentParser.py

```diff
@@ -289,13 +290,15 @@ def testAtFile(self):
     def testLogDest(self):
         """Test --logdest
         """
-        logFile = "tests_argumentParser_testLog_temp.txt"
-        self.ap.parse_args(
-            config = self.config,
-            args = [DataPath, "--logdest", logFile],
-        )
-        self.assertTrue(os.path.isfile(logFile))
-        os.remove(logFile)
```

Comment:
If I change the `utils` leak checker to use `lsof` then this reports a leak as expected. I'm not entirely sure how to proceed. The new patch is not correct because it would work even if the logger failed to do anything. The old code really does leak a file descriptor and the leak checker in `psutil` is not really helping. Do I migrate `utils` to use `lsof` in preference to `psutil`?